rust_icu_ecma402/src/numberformat.rs

```diff
@@ -0,0 +1,351 @@
+                // Nothing is needed here.
+            }
+            options::Notation::Engineering => match opts.sign_display {
+                options::SignDisplay::Auto => {
+                    skel.push(format!("scientific/*ee"));
+                }
+                options::SignDisplay::Always => {
+                    skel.push(format!("scientific/*ee/sign-always"));
+                }
+                options::SignDisplay::Never => {
+                    skel.push(format!("scientific/*ee/sign-never"));
+                }
+                options::SignDisplay::ExceptZero => {
+                    skel.push(format!("scientific/*ee/sign-expect-zero"));
+                }
+            },
+            options::Notation::Scientific => {
+                skel.push(format!("scientific"));
+            }
+            options::Notation::Compact => {
+                // ?? Is this true?
+                skel.push(format!("compact-short"));
+            }
+        }
+        if let Some(ref n) = opts.numbering_system {
+            skel.push(format!("numbering-system/{}", &n.0));
+        }
+
+        if opts.notation != options::Notation::Engineering {
+            match opts.sign_display {
+                options::SignDisplay::Auto => {
+                    skel.push("sign-auto".into());
+                }
+                options::SignDisplay::Never => {
+                    skel.push("sign-never".into());
+                }
+                options::SignDisplay::Always => {
+                    skel.push("sign-always".into());
+                }
+                options::SignDisplay::ExceptZero => {
+                    skel.push("sign-always".into());
+                }
+            }
+        }
+
+        let minimum_integer_digits = opts.minimum_integer_digits.unwrap_or(1);
+        // TODO: this should match the list at:
+        // https://www.currency-iso.org/en/home/tables/table-a1.html
+        let minimum_fraction_digits = opts.minimum_fraction_digits.unwrap_or(match opts.style {
+            options::Style::Currency => 2,
+            _ => 0,
+        });
+        let maximum_fraction_digits = opts.maximum_fraction_digits.unwrap_or(match opts.style {
+            options::Style::Currency => std::cmp::max(2, minimum_fraction_digits),
+            _ => 3,
+        });
+        let minimum_significant_digits = opts.minimum_significant_digits.unwrap_or(1);
+        let maximum_significant_digits = opts.maximum_significant_digits.unwrap_or(21);
+
+        // TODO: add skeleton items for min and max integer, fraction and significant digits.
+        skel.push(integer_digits(minimum_integer_digits as usize));
+        skel.push(fraction_digits(
+            minimum_fraction_digits as usize,
+            maximum_fraction_digits as usize,
+            minimum_significant_digits as usize,
+            maximum_significant_digits as usize,
+        ));
+
+        Ok(skel.iter().map(|s| format!("{} ", s)).collect())
```

Comment:
Is there supposed to be a trailing space?
If not, you could use `skel.join(" ")` here.

Reply:
I was slightly lazy to sanitize this last space, and didn't want to use `join` as it seems to be an experimental API. https://doc.rust-lang.org/std/slice/trait.Join.html#tymethod.join

I am not too worried since this value is never exposed to the user so can be changed at any time we so decide.